pkg/datastore/filedb/filter.go

```diff
@@ -15,19 +15,181 @@
 package filedb
 
 import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+	"unicode"
+
 	"github.com/pipe-cd/pipecd/pkg/datastore"
 )
 
-// TODO: Implement filterable interface for each collection.
 type filterable interface {
 	Match(e interface{}, filters []datastore.ListFilter) (bool, error)
 }
 
 func filter(col datastore.Collection, e interface{}, filters []datastore.ListFilter) (bool, error) {
+	// If the collection implement filterable interface, use it.
 	fcol, ok := col.(filterable)
-	if !ok {
+	if ok {
+		return fcol.Match(e, filters)
+	}
+
+	// remarshal entity as map[string]interface{} struct.
+	raw, err := json.Marshal(e)
+	if err != nil {
+		return false, err
+	}
+	var omap map[string]interface{}
+	if err := json.Unmarshal(raw, &omap); err != nil {
+		return false, err
+	}
+
+	for _, filter := range filters {
+		field := convertCamelToSnake(filter.Field)
+		if strings.Contains(field, ".") {
+			// TODO: Handle nested field name such as SyncState.Status.
+			return false, datastore.ErrUnsupported
+		}
+
+		val, ok := omap[field]
+		// If the object does not contain given field name in filter, return false immidiately.
+		if !ok {
+			return false, nil
+		}
+
+		cmp, err := compare(val, filter.Value, filter.Operator)
+		if err != nil {
+			return false, err
+		}
+
+		if !cmp {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+func compare(val, operand interface{}, op datastore.Operator) (bool, error) {
+	var valNum, operandNum int64
+	switch v := val.(type) {
+	case int, int8, int16, int32, int64:
+		valNum = reflect.ValueOf(v).Int()
+	case uint, uint8, uint16, uint32:
+		valNum = reflect.ValueOf(v).Int()
```

Comment:
```suggestion
		valNum = int64(reflect.ValueOf(v).Uint())
```
Sorry for the lack of explanation. 
if the underlying type is `uint`, `Int()` will be panic(https://pkg.go.dev/reflect#Value.Int). So calling `Uint()` and then casting to int is better.

Reply:
oops, thanks for explanation 👍